File: src/test/java/org/oewntk/parse/TestDomains.java
/*
 * Copyright (c) 2021. Bernard Bou.
 */

package org.oewntk.parse;

import org.junit.Test;
import org.oewntk.pojos.Domain;
import org.oewntk.pojos.ParsePojoException;
import org.oewntk.utils.Tracing;

import java.io.PrintStream;

public class TestDomains
{
	private static final PrintStream ps = !System.getProperties().containsKey("SILENT") ? Tracing.psInfo : Tracing.psNull;

	@Test
	public void testDomains() throws ParsePojoException
	{
		for (Domain domain : Domain.getEntries())
		{
			String domainStr = domain.getDomain();
			ps.printf("%d %s %s %s %s%n", domain.id, domain.getName2(), domain.pos, domainStr, Domain.parseDomain(domainStr));
		}
	}

	@Test
	public void testDomainsIds() throws ParsePojoException
	{
		for (Domain domain : Domain.getEntries())
		{
			String domainId = Integer.toString(domain.id);
			Domain domain2 = Domain.parseDomainId(domainId);
			ps.printf("%s -> %s %s%n", domainId, domain2.getName2(), domain2.pos);
		}
	}
}

File: src/test/java/org/oewntk/parse/TestDomains.kt
/*
 * Copyright (c) 2021-2024. Bernard Bou.
 */
package org.oewntk.parse

import org.junit.Test
import org.oewntk.pojos.Domain
import org.oewntk.pojos.Domain.Companion.parseDomain
import org.oewntk.pojos.Domain.Companion.parseDomainId
import org.oewntk.pojos.ParsePojoException
import org.oewntk.utils.Tracing

class TestDomains {

    @Test
    @Throws(ParsePojoException::class)
    fun testDomains() {
        Domain.entries.forEach {
            ps.println("${it.id} ${it.name2} ${it.pos} ${it.domain} ${parseDomain(it.domain)}")
        }
    }

    @Test
    @Throws(ParsePojoException::class)
    fun testDomainsIds() {
        Domain.entries.forEach {
            val domain2 = parseDomainId(it.domain)
            ps.println("${it.id} -> ${domain2.name2} ${domain2.pos}")
        }
    }

    companion object {

        private val ps = if (!System.getProperties().containsKey("SILENT")) Tracing.psInfo else Tracing.psNull
    }
}
